.github/workflows/ci.yaml

```diff
@@ -18,16 +18,20 @@ jobs:
         uses: actions/setup-go@v2
         with:
           go-version: ${{ matrix.go-version }}
+          
       - name: Set up a Vertica server
-        env:
-          VERTICA_CE_URL: "https://vertica-community-edition-for-testing.s3.amazonaws.com/XCz9cp7m/vertica-12.0.0-0.x86_64.RHEL6.rpm"
         run: |
-          git clone https://github.com/jbfavre/docker-vertica.git
-          curl $VERTICA_CE_URL --create-dirs -o docker-vertica/packages/vertica-ce.latest.rpm
-          docker build -f docker-vertica/Dockerfile.centos.7_9.x --build-arg VERTICA_PACKAGE=vertica-ce.latest.rpm -t jbfavre/vertica docker-vertica
-          docker images
-          docker run --hostname vertica-docker --name vertica_docker -d -p 5433:5433 jbfavre/vertica
-          sleep 60
+          docker pull vertica/vertica-ce:12.0.0-0
+          docker run -d -p 5433:5433 -p 5444:5444 \
+            --name vertica_docker \
+            vertica/vertica-ce
+          echo "Vertica startup ..."
+          until docker exec vertica_docker /opt/vertica/bin/vsql -c "SELECT 1" > /dev/null 2>&1; do \
```

Comment:
The nondeterministic failures comes from this check. During the docker container startup, it first creates a database (`SELECT 1` works at this stage), then loads VMart data and start agent. The whole process takes about 90 secs. This causes the golang tests start too early and conflict with container startup.

Reply:
nice catch, thanks for figuring this out.